snapcraft/storeapi/_upload.py

```diff
@@ -125,9 +161,8 @@ def upload_files(binary_filename, config=None):
             'An unexpected error was found while uploading files.')
         result['errors'] = [str(err)]
     finally:
-        # make sure to close any open files used for request
-        for fd in files.values():
-            fd.close()
+        # Close the open file
+        binary_file.close()
```

Comment:
@sergiusens copied the same block in #285. I prefer your removal of the confusing dictionary of 1 element, so I think this one should land first.